### DISCUSSION TITLE
使用Poetry构建Bot时，安装插件nb plugin install 或 poetry add？

### DISCUSSION THREAD

**Discussion by KafCoppelia** [nonebot/discussions#83] [想法]
使用Poetry构建Bot时，安装插件商店里的插件如何安装比较好，是`nb plugin install`还是`poetry add`作为`[tool.poetry.dependencies]
`？
如果使用前者，前者的安装源代码不好找，我可能有更改插件源码的需求；选后者，`poetry add`，插件会位列在`tool.poetry.dependencies`内，这和自己编写的插件（文件夹位于`/botname/plugin`下的这种，导入bot后列在`pyproject.toml`下的`[tool.nonebot]`内），同是插件类别却在不同的种类下，有点怪。

**Comment by mnixry** [comment]
> 如果使用前者，前者的安装源代码不好找，我可能有更改插件源码的需求

事实上, 无论是上述的任何一种指令安装, 都不能符合自己能随时修改源代码的需求
在有自己修改插件源代码的需求的情况下, 您不应该使用`nb plugin install`或者`poetry add`这其中的任何一个, 而是考虑将该插件的源码下载下来整合进您的项目
 
> 同是插件类别却在不同的种类下，有点怪

`poetry add`是将该插件加入项目依赖, 在以后执行`poetry install`的时候会一并安装, 而加入`[tool.nonebot]`则只是向NoneBot声明你需要在NoneBot内加载该插件, 这两个项目的作用是不同的

**Comment by KafCoppelia** [comment]
Got it！🥳
